stages/org.osbuild.ostree.deploy.container

```diff
@@ -105,10 +107,62 @@ def ostree_container_deploy(tree, inputs, osname, target_imgref, kopts):
             extra_args.append(f'--target-imgref={target_imgref}')
 
             kargs = [f'--karg={v}' for v in kopts]
-
             ostree.cli("container", "image", "deploy",
                        *extra_args, sysroot=tree, *kargs)
 
+            # the `ostree container image metadata` command utilizes the
+            # target-imgref, but only takes the name of the container image
+            target_imgref_list = target_imgref.split(':')
+            if target_imgref_list[0] == "ostree-remote-registry":
+                target_imgref_name = ':'.join(target_imgref_list[2:])
+            elif target_imgref_list[0] == ("ostree-image-signed" or "ostree-unverified-registry"):
+                target_imgref_name = ':'.join(target_imgref_list[1:])
+            
+            container_data = subprocess.run(["ostree", "container", "image", "metadata", f"--repo={tree}/ostree/repo", f"registry:{target_imgref_name}"],
+					check=True,
+					capture_output=True,
+					text=True).stdout.rstrip()
+            
+            container_data_config = subprocess.run(["ostree", "container", "image", "metadata", f"--repo={tree}/ostree/repo", f"registry:{target_imgref_name}", "--config"],
+					check=True,
+					capture_output=True,
+					text=True).stdout.rstrip()
+            
+            container_data_json = json.loads(container_data)
+            container_data_config_json = json.loads(container_data_config)
+
+            aleph_digest = container_data_json['config']['digest']
+            aleph_ref = f"docker://{target_imgref}"
+            aleph_version = container_data_config_json['config']['Labels']['org.opencontainers.image.version']
+            aleph_platform = "" # still up for discussion
+            aleph_imgid = "" # still up for discussion
+            aleph_container_image = container_data_config_json['config']['Labels']
+
+            aleph_version_data = {
+                "osbuild-version": osbuild.__version__,
+                "ref": aleph_ref,
+                "build": aleph_version,
+                "version": aleph_version,
+                "platform": aleph_platform,
+                "imgid": aleph_imgid,
+                "container-image": {
+                    "image-name": image_name,
+                    "image-digest": aleph_digest,
+                    "image-labels": aleph_container_image
+                }
+            }
+            
+            # the 'ostree.commit' label will be optional in the future so
+            # prevent hard failing if key is not found
+            aleph_ostree_commit = container_data_config_json['config']['Labels'].get('ostree.commit')
+            if aleph_ostree_commit is not None:
+                aleph_version_data["ostree-commit"] = aleph_ostree_commit
+
+            with open(tree + "/.osbuild-container-aleph.json", "w") as f:
```

Comment:
I'm still not sure what the right answer is here for the name of this file. Obviously coming from where we are right now with coreos I want to keep it named `.coreos-aleph-version.json` but I think that's only because I don't know of a good "one size fits all" answer here. Though I do think we need to keep backwards compat (maybe a symlink) with `.coreos-aleph-version.json` for continuity for FCOS and RHCOS.

in the future calling it `.osbuild-container-aleph.json` feels a little weird. I'm sure other things *could* build images. Maybe just `aleph-version.json` and we stick in there a field that lets the consumer know what tool created the aleph version file?

Reply:
@cgwalters  @jlebon - WDYT?

Reply:
I'm fine with whatever because honestly I do want to replace this stuff with a bootc stage in the future in which case it will use the bootc name :smile: